string question
uint8 sys_state
